Lib/asyncio/streams.py

```diff
@@ -571,6 +571,11 @@ def readuntil(self, separator=b'\n'):
                 self._buffer.clear()
                 raise IncompleteReadError(chunk, None)
```

Comment:
I don't like the fact that this code path can swallow the previously set `self._exception`.